pptx_generator/src/handlers/generate_pptx.py

```diff
@@ -72,490 +36,29 @@ def get_commands_list(self):
         return self.commands.keys()
 
     def get_command(self, command_name):
-        if command_name == Command.REPLACE_IMAGE:
+        if command_name == Command.REPLACE_IMAGE.value:
             image = Image()
-            def getParams(commands_dic, presentation, slide, shape, slides, dataObj):
-               return image.replace_images(slide, CommandRegex.IMAGE.value[0], shape)
-            return getParams
-
-        elif command_name == Command.TEXT_REPLACE:
+            return image.replace_images
+         
+        elif command_name == Command.TEXT_REPLACE.value:
             text = Text()
-            def getParams(commands_dic, presentation, slide, shape, slides, dataObj):
-                return text.text_replace(slide, CommandRegex.TEXT.value[0], shape)
-            return getParams
-        
-        elif command_name == Command.UPDATE_TABLE_TEXT:
+            return text.text_replace
+            
+        elif command_name == Command.UPDATE_TABLE_TEXT.value:
             table = Table()
             return table.update_table_text
           
-        elif command_name == Command.DRAW_TABLE:
+        elif command_name == Command.DRAW_TABLE.value:
             table = Table()
             return table.drow_tables
-            
+
+        elif command_name == Command.IF_CONDITION:
+                expression = Expression(Tag)
```

Comment:
Don't pass Tag class as an argument. Use methods in the Tag class by using inheritance in the Expression class.